In [24]:
import pandas as pd
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, confusion_matrix

In [3]:
data = pd.read_csv("./data/train.csv")
useful_features = ["age", "sex", "cp", "trestbps", "chol", "fbs", "thalach", "exang"]
X = data[useful_features]
Y = data["target"]

In [9]:
X_train, X_valid, y_train, y_valid = train_test_split(X, Y, test_size=0.2)

In [10]:
X_train.shape, y_train.shape

((656, 8), (656,))

### Decision Tree model

In [11]:
decision_model = DecisionTreeClassifier(random_state=0)
decision_model.fit(X_train, y_train)

DecisionTreeClassifier(random_state=0)

In [12]:
decision_valid_preds = decision_model.predict(X_valid)
decision_valid_preds[:5]


array([1, 1, 1, 0, 0], dtype=int64)

In [17]:
decision_model.score(X_valid, y_valid)

0.9817073170731707

In [16]:
confusion_matrix(y_valid, decision_valid_preds)

array([[77,  1],
       [ 2, 84]], dtype=int64)

In [26]:
print(classification_report(y_valid, decision_valid_preds))

              precision    recall  f1-score   support

           0       0.97      0.99      0.98        78
           1       0.99      0.98      0.98        86

    accuracy                           0.98       164
   macro avg       0.98      0.98      0.98       164
weighted avg       0.98      0.98      0.98       164



accuracy is already 98% and f1 score is also 98%, that's aweasome.

A simple decision tree worked a lot better